### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Fixes several minor documentation issues.

# Details

Fixes several typos (missing new line or a : rather than ::) and
tweaks the text in several methods, as well as adding ReSampleData to the RTD documentation. Several routines added in #868 had their documentation tweaked: noting they were new and adding them to the RTD documentation.

### CODE REVIEW

**Comment by DougBurke** [comment]
@anetasie - this is a docs-only change, so I've requested your review. It improves the RTD site but is also needed for the ahelp files

**Comment by anetasie** [comment]
@DougBurke I left two comments. The other updates look good.

File: sherpa/sim/__init__.py [hunk 1]
```diff
@@ -785,11 +785,21 @@ class ReSampleData(NoNewAttributesAfterInit):
 
     Returns
     -------
-    The class returns the best fit parameters for each realization.
+    sampled : dict
+       The keys are samples, which contains the resampled data
+       used in the fits as a niter by ndata array, and the free
+       parameters in the fit, containing a NumPy array containing
+       the fit parameter for each iteration (of size niter).
```
[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:792]: Actually, no.

[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:792]: resample_data doesn't fit at each iteration, it just creates a set of randomly drawn values and you get those values and the statistic value

[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:792]: Oh. I've got myself confused. Too many files being edited. I'm goiing back to my original "yes".

File: sherpa/sim/__init__.py [hunk 2]
```diff
@@ -785,11 +785,21 @@ class ReSampleData(NoNewAttributesAfterInit):
 
     Returns
     -------
-    The class returns the best fit parameters for each realization.
+    sampled : dict
```
[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:788]: Fixed now.ta.

File: sherpa/sim/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (anetasie) [reply sherpa/sim/__init__.py:792]: I think the resampled data should be fit. Can you confirm, as now I'm confused too.

**Comment by anetasie** [comment]
@DougBurke  I left one more comment.

File: sherpa/sim/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:792]: It is - you can see the fit call at https://github.com/sherpa/sherpa/blob/74aba2094359f5d65248b2ad2ea5116fded04770/sherpa/sim/__init__.py#L958

[the remark this answers is not in the document]
Comment (DougBurke) [reply sherpa/sim/__init__.py:792]: And the later calls show that the best-fit parameter from each iteration is returned.

File: sherpa/sim/__init__.py [hunk 3]
```diff
@@ -785,11 +785,21 @@ class ReSampleData(NoNewAttributesAfterInit):
...
+       The keys are samples, which contains the resampled data
+       used in the fits as a niter by ndata array, and the free
+       parameters in the fit, containing a NumPy array containing
```
Comment (anetasie) [inline sherpa/sim/__init__.py:791]: Maybe add 'best-fit' parameters in the fit?

**Comment by anetasie** [comment]
@DougBurke I just added a comment suggesting to add 'best-fit' into the docstring.

File: sherpa/sim/__init__.py (hunk 3 above)
Comment (DougBurke) [reply sherpa/sim/__init__.py]: I've just made an update. It is now

https://github.com/sherpa/sherpa/blob/c3de99e5136f75a0e6829942c6c94887a8fdff87/sherpa/sim/__init__.py#L788-L793

**Review by anetasie — APPROVED** [review]

**Pull request merged**